public/app/core/components/TransformersUI/lookupGazetteer/fieldLookup.ts

```diff
@@ -44,21 +53,22 @@ export function addFieldsFromGazetteer(frames: DataFrame[], gaz: Gazetteer, matc
       //if the field matches
       if (matcher(field, frame, frames)) {
         const values = field.values.toArray();
-        const lat = new Array<Number>(values.length);
-        const lon = new Array<Number>(values.length);
-
-        //for each value find the corresponding value in the gazetteer
-        for (let v = 0; v < values.length; v++) {
-          const foundMatchingValue = gaz.find(values[v]);
+        const sub: any[][] = [];
+        for (const f of src) {
+          const buffer = new Array(length);
+          sub.push(buffer);
+          fields.push({ ...f, values: new ArrayVector(buffer) });
+        }
 
-          //for now start by adding lat and lon
-          if (foundMatchingValue && foundMatchingValue?.coords.length) {
-            lon[v] = foundMatchingValue.coords[0];
-            lat[v] = foundMatchingValue.coords[1];
+        // Add all values to the buffer
+        for (let v = 0; v < sub.length; v++) {
+          const found = gaz.find(values[v]);
+          if (found?.index != null) {
+            for (let i = 0; i < src.length; i++) {
+              sub[i][v] = src[i].values.get(found.index);
```

Comment:
Are we using this sub matrix anywhere? It doesn't seem to be attached to any object we are returning currently so I'm not sure its purpose

Reply:
it is from an alpha panel that is not exposed yet -- but the test is.  This will get a lot of attention in the next PR :)